zerver/models.py

```diff
@@ -402,6 +402,12 @@ class Realm(models.Model):
     STANDARD_FREE = 4
     plan_type: int = models.PositiveSmallIntegerField(default=SELF_HOSTED)
 
+    # Who in the organization is allowed to create streams.
+    OSS_SHOWCASE_HIDDEN = 1
+    OSS_SHOWCASE_LISTED = 2
+    oss_showcase_policy: int = models.PositiveSmallIntegerField(default=OSS_SHOWCASE_HIDDEN)
+    OSS_SHOWCASE_POLICY_TYPES = [OSS_SHOWCASE_HIDDEN, OSS_SHOWCASE_LISTED]
```

Comment:
Having thought about naming a bit more, let's name these

CUSTOMER_SHOWCASE_* rather than OSS_SHOWCASE_*; we'll likely end up wanting to use additional values for this setting for "Classes using Zulip" as well, which might have different semantics at the showcase layer.